e2e/test/scenarios/custom-column/custom-column-reproductions.cy.spec.js

```diff
@@ -1055,69 +1053,83 @@ describe("issue 49882", () => {
   it("should not eat up subsequent characters when applying a suggestion (metabase#49882-1)", () => {
     const moveCursorTo2ndCaseArgument = "{leftarrow}".repeat(6);
     H.enterCustomColumnDetails({
-      formula: `case([Total] > 200, , "X")${moveCursorTo2ndCaseArgument}[tot{enter}`,
+      formula: `case([Total] > 200, , "X")${moveCursorTo2ndCaseArgument}[tot`,
+      blur: false,
     });
 
-    cy.get(".ace_text-input")
-      .first()
-      .should("have.value", 'case([Total] > 200, [Total] , "X")\n\n');
+    H.CustomExpressionEditor.completions().should("be.visible");
+    cy.realPress("Enter", { pressDelay: 10 });
+
+    H.CustomExpressionEditor.shouldContain('case([Total] > 200, [Total], "X")');
     H.popover()
       .findByText("Expecting a closing parenthesis")
       .should("not.exist");
   });
 
   it("does not clear expression input when expression is invalid (metabase#49882-2, metabase#15892)", () => {
-    const selectTax = `{leftarrow}${"{shift+leftarrow}".repeat(5)}`;
-    const moveCursorBefore2ndCase = "{leftarrow}".repeat(41);
+    // This test used to use keyboard shortcuts to cut and paste but this
+    // seem impossible to emulate with CodeMirror in Cypress, so it's using allowFastSet
+    // instead.
+
     H.enterCustomColumnDetails({
-      formula: `case([Tax] > 1, case([Total] > 200, [Total], "Nothing"), [Tax])${selectTax}`,
+      formula:
+        'case([Tax] > 1, case([Total] > 200, [Total], "Nothing"), [Tax])',
+      blur: false,
+      allowFastSet: true,
     });
-    cy.get(".ace_text-input").first().focus().realPress(["Control", "X"]);
-    cy.get(".ace_text-input")
-      .first()
-      .focus()
-      .type(moveCursorBefore2ndCase)
-      .realPress(["Control", "V"]);
-    cy.get(".ace_text-input").first().focus().type(" ").blur();
-
-    cy.get(".ace_text-input")
-      .first()
-      .focus()
-      .should(
-        "have.value",
-        'case([Tax] > 1, [Tax] case([Total] > 200, [Total], "Nothing"), )\n\n',
-      );
+
+    // Cut [Tax]
+    H.CustomExpressionEditor.type(
+      'case([Tax] > 1, case([Total] > 200, [Total], "Nothing"), )',
+      {
+        allowFastSet: true,
+      },
+    );
+
+    // Paste [Tax] before case
+    H.CustomExpressionEditor.type(
+      'case([Tax] > 1, [Tax] case([Total] > 200, [Total], "Nothing"), )',
+      {
+        allowFastSet: true,
+      },
+    );
+
+    H.CustomExpressionEditor.shouldContain(
+      'case([Tax] > 1, [Tax] case([Total] > 200, [Total], "Nothing"), )',
+    );
+    H.CustomExpressionEditor.blur();
 
     H.popover()
       .findByText("Expecting comma but got case instead")
       .should("be.visible");
   });
 
-  it("should allow moving cursor between wrapped lines with arrow up and arrow down keys (metabase#49882-3)", () => {
+  // TODO: we no longer have wrapped lines (for now)
```

Comment:
Let's make sure we have an issue for this

Reply:
This is part of the project I'm working on: we're moving to a multiline editor. (The relevant issue is QUE-582).